Project-Euler/Project-Euler/Problems.cs

```diff
@@ -87,11 +87,11 @@ public static void Problem3()
             Console.WriteLine($"\n\tSolution: {numberToBeDivided}");
         }
 
-        private static bool IsItPrime(long number)
+        private static bool IsPrime(long number)
         {
-            int squareRoot = (int)Math.Sqrt(number);
+            long squareRoot = (long)Math.Sqrt(number);
 
-            for (int i = 2; i < squareRoot; i++)
+            for (int i = 2; i <= squareRoot; i++)
```

Comment:
You could make this much more performant by checking if it's divisible by 2 before the loop and then only looping over the odd numbers. This should cut the time in half!

Reply:
Also nice catch on the bug!